src/FastConv/FastConv.cpp

```diff
@@ -10,23 +11,259 @@ CFastConv::~CFastConv( void )
+        for (int i=0; i<iBlockLength*(m_iBlockNum+1); i++)
+            m_pfConvMatrix[c][i] = 0.f;
+    }
+    
+    m_pfConvOuput = new float [m_iBlockNum*m_iBlockLength];
+    for (int i=0; i<m_iBlockNum*m_iBlockLength; i++)
+        m_pfConvOuput[i] = 0.f;
+    
+    
+    m_bIsInitialized = true;
 
     return Error_t::kNoError;
 }
 
 Error_t CFastConv::reset()
 {
+    if(m_bIsInitialized)
+    {
+        for (int c=0; c<m_iBlockNum+1; c++)
+            delete[] m_pfIRMatrix[c];
+        delete m_pfIRMatrix;
+        
+        for (int c=0; c<m_iBlockNum+1; c++)
+            delete[] m_pfConvMatrix[c];
+        delete m_pfConvMatrix;
+        
+        delete[] m_pfInputBlock;
+        delete[] m_pfOutputBlock;
+        delete[] m_pfBlockConvOuput;
+        delete[] m_pfConvOuput;
+        
+        m_iBlockNum = 0;
+        m_iCirIdx = 0;
+        m_iBlockLength = 0;
+        m_iLengthOfIr = 0;
+    }
+    
+    m_bIsInitialized = false;
+    
+    return Error_t::kNoError;
+}
+
+
+Error_t CFastConv::timeConv(int idx)
+{
+    for (int i=0; i<2*m_iBlockLength-1; i++)
+    {
+        float sum = 0.f;
+        for (int m = 0; m < 2*m_iBlockLength; m++)
+        {
+            sum += m_pfInputBlock[m] * m_pfIRMatrix[idx][i - m];
+        }
+        m_pfBlockConvOuput[i] = sum;
+    }
+    return Error_t::kNoError;
+}
+
+
+Error_t CFastConv::freqConv(int idx)
+{
+    m_pfTimeIr = m_pfIRMatrix[idx];
+    // Take FFT of x and h, store in m_pfFreqInput, m_pfFreqIr
+    m_pCFftInstance->doFft(m_pfFreqInput, m_pfTimeInput);
+    m_pCFftInstance->doFft(m_pfFreqIr, m_pfIRMatrix[idx]);
+    // Split the real and imaginary parts for multiplication
+    m_pCFftInstance->splitRealImag(m_pfRealInput, m_pfImagInput, m_pfFreqInput);
+    m_pCFftInstance->splitRealImag(m_pfRealIr, m_pfImagIr, m_pfFreqIr);
+
+    // frequency domain multiplication == time domain convolution
+    for (int i = 0; i < (2 * m_iBlockLength); i++)
+    {
+        float ac =  m_pfRealInput[i] * m_pfRealIr[i];
+        float bc = m_pfImagInput[i] * m_pfRealIr[i];
+        float ad = m_pfRealInput[i] * m_pfRealIr[i];
+        float bd = m_pfImagInput[i] * m_pfImagIr[i];
+        m_pfRealConv[i] = ac - bd;
+        m_pfImagConv[i] = ad + bc;
+    }
+    m_pCFftInstance->mergeRealImag(m_pfFreqConv, m_pfRealConv, m_pfImagConv);
+    // Convert back to time domain
+    m_pCFftInstance->doInvFft(m_pfConvOuput, m_pfFreqConv);
+}
+
+
+
+
+Error_t CFastConv::conv()
+{
+    //clear previous values in m_pfConvMatrix
+    for (int i=0; i<m_iBlockLength*(m_iBlockNum+1); i++)
+        m_pfConvMatrix[m_iCirIdx][i] = 0.f;
+    
+    for (int i=0; i<m_iBlockLength; i++)
+        m_pfOutputBlock[i] = 0.f;
+    
+    for (int i=0; i<m_iBlockLength*2; i++)
+        m_pfBlockConvOuput[i] = 0.f;
+    
+    
+    for (int i = 0; i<m_iBlockNum; i++)
+    {
+        //switch to freqConv() after implemented
+        switch (m_eCompMode) {
+            case kTimeDomain:
+                timeConv(i);
```

Comment:
It looks like you're doing the same kind of overlap-add thing in the time-domain mode as in the frequency-domain mode. This is fine for testing, but it's no more efficient than ordinary time-domain convolution (using the whole impulse response), and once your implementation is complete you may find that it adds unnecessary latency to the time-domain convolution output.